### PR TITLE
Fixed add ssh key to cluster

### PR DESCRIPTION
**Pull request by glower**
**What this PR does / why we need it**:

Fixed adding ssh keys to the project. Now we use a new PUT route for this. This PR will be pushed together with https://github.com/kubermatic/dashboard-v2/pull/791 where frontend is fixed to use new PUT route

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Documentation**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: api/pkg/handler/cluster.go [hunk 1]
```diff
@@ -624,14 +623,24 @@ func prometheusQueryRange(ctx context.Context, api prometheusv1.API, query strin
...
 	// in: path
 	ClusterID string `json:"cluster_id"`
+	// in: path
+	KeyID string `json:"key_id"`
 }
 
-type assignSSHKeysToClusterBodyReq struct {
-	KeyName string `json:"KeyName"`
-}
+// AssignSSHKeysToClusterReq defines HTTP request data for newAssignSSHKeyToCluster  endpoint
```
Comment (p0lyn0mial) [inline api/pkg/handler/cluster.go]: rm commented code.

File: api/pkg/handler/cluster_test.go [hunk 2]
```diff
@@ -659,6 +679,32 @@ func TestAssignSSHKeyToClusterEndpoint(t *testing.T) {
 			}
 
 			compareWithResult(t, res, tc.ExpectedResponse)
+
+			kubermaticClient := clientsSets.fakeKubermaticClient
+			validatedActions := 0
+			if tc.HTTPStatus == http.StatusCreated {
+				for _, action := range kubermaticClient.Actions() {
+					fmt.Printf("%s %s\n", action.GetResource(), action.GetVerb())
```
Comment (p0lyn0mial) [inline api/pkg/handler/cluster_test.go]: rm this line.

File: api/pkg/handler/cluster_test.go [hunk 3]
```diff
@@ -659,6 +679,32 @@ func TestAssignSSHKeyToClusterEndpoint(t *testing.T) {
...
+						}
+						for _, expectedSSHKey := range tc.ExpectedSSHKeys {
+							sshKeyFromAction := updateAction.GetObject().(*kubermaticv1.UserSSHKey)
+							if sshKeyFromAction.Name == expectedSSHKey.Name {
+								if !equality.Semantic.DeepEqual(updateAction.GetObject().(*kubermaticv1.UserSSHKey), expectedSSHKey) {
+									t.Fatalf("%v", diff.ObjectDiff(expectedSSHKey, updateAction.GetObject().(*kubermaticv1.UserSSHKey)))
+								}
+							}
+						}
+						validatedActions = validatedActions + 1
```
Comment (p0lyn0mial) [inline api/pkg/handler/cluster_test.go]: shouldn't this be inside the if statement ? (`sshKeyFromAction.Name == expectedSSHKey.Name`)

File: api/pkg/handler/cluster.go [hunk 4]
```diff
@@ -325,7 +325,7 @@ func assignSSHKeyToCluster(sshKeyProvider provider.NewSSHKeyProvider, projectPro
 			return nil, kubernetesErrorToHTTPError(err)
 		}
 
-		_, err = clusterProvider.Get(userInfo, req.ClusterID, &provider.ClusterGetOptions{CheckInitStatus: true})
+		_, err = clusterProvider.Get(userInfo, req.ClusterID, &provider.ClusterGetOptions{CheckInitStatus: false})
```
Comment (p0lyn0mial) [inline api/pkg/handler/cluster.go:328]: may I know what does it change ?

File: api/pkg/handler/cluster_test.go (hunk 3 above)
Comment (glower) [reply api/pkg/handler/cluster_test.go]: Yes, you right

File: api/pkg/handler/cluster_test.go (hunk 2 above)
Comment (glower) [reply api/pkg/handler/cluster_test.go]: done

File: api/pkg/handler/cluster.go (hunk 1 above)
Comment (glower) [reply api/pkg/handler/cluster.go]: done

File: api/pkg/handler/cluster.go (hunk 4 above)
Comment (glower) [reply api/pkg/handler/cluster.go:328]: This doesn't check if the cluster is already healthy.  Otherwise, the code will respond with 503 or something.

Comment (p0lyn0mial) [reply api/pkg/handler/cluster.go:328]: oh, nice :)

**Review by p0lyn0mial — APPROVED** [review]

**Pull request merged**